source/write/transaction.txt

```diff
@@ -0,0 +1,183 @@
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to use the {+php-library+} to perform
+**transactions**. Transactions allow you to perform a series of operations
+that change data only if the entire transaction is committed.
+If any operation in the transaction does not succeed, the library stops the
+transaction and discards all data changes before they ever become
+visible. This feature is called **atomicity**.
+
+In MongoDB, transactions run within logical sessions. A
+session is a grouping of related read or write operations that you
+want to run sequentially. Sessions enable causal consistency for a group
+of operations and allow you to run operations in an **ACID-compliant**
+transaction, which is a transaction that meets an expectation of
+atomicity, consistency, isolation, and durability. MongoDB guarantees
+that the data involved in your transaction operations remains
+consistent, even if the operations encounter unexpected errors.
+
+When using the {+php-library+}, you can create a new session from a
+``MongoDB\Client`` instance. Then, you can use the resulting
+``MongoDB\Driver\Session`` instance to perform transactions. You can
+improve your app's performance by reusing your client for multiple
+sessions and transactions instead of instantiating a new client each
+time.
+
+.. warning::
+
+   Use a ``Session`` only in operations running on the
+   ``Client`` that created it. Using a ``Session`` with a
+   different ``Client`` results in operation errors.
+
+Methods
+-------
+
+Create a ``Session`` by using the ``MongoDB\Client::startSession()``
+method on your ``Client`` instance. The {+php-library+} provides a
+Convenient Transaction API to manage the transaction lifecyle. Use the
+``MongoDB\with_transaction()`` method to run custom callback within a
+transaction. The ``with_transaction()`` method starts the transaction,
+then either commits it or ends it if there are errors. The
+:ref:`php-txn-example` section of this guide demonstrates how to use
+this API to perform a transaction. 
+
+Alternatively, you can have more control over your transaction lifecyle
+by using the methods provided by the ``Session`` class. The
+following table describes these methods:
+
+.. list-table::
+   :widths: 25 75
+   :stub-columns: 1
+   :header-rows: 1
+
+   * - Method
+     - Description
+
+   * - ``startTransaction()``
+     - | Starts a new transaction on this session. The session
+         must be passed into each operation within the transaction, or
+         the operation will run outside of the transaction.
+       |
+       | You can set transaction options by passing an options parameter.
+
+   * - ``commitTransaction()``
+     - | Commits the active transaction for this session. This method returns an
+         error if there is no active transaction for the session or the
+         transaction was previously ended.
```

Comment:
Notably, a commit can fail due to other transient errors, such as a write conflict. That would be more common as a runtime error, vs. a user calling this when there is no active transaction (likely a poorly written application, and a mistake they'd make once and correct).

Reply:
added that clause